internal/services/fingerprint/consumer.go

```diff
@@ -148,6 +150,8 @@ func (c *Consumer) HandleDeviceFingerprint(ctx context.Context, event *Event) er
 			c.logger.Error().Msgf("could not marshal userdevice metadata for device: %s", ad.R.VehicleToken.R.UserDevice.ID)
 			return err
 		}
+
+		appmetrics.FingerprintRequestCount.With(prometheus.Labels{"protocol": *md.CANProtocol}).Inc()
```

Comment:
there could be a case where this would nil pointer dereference. Consider using the `protocol` variable above instead (returned by ExtractProtocol)